torch/_inductor/ir.py

```diff
@@ -5567,3 +5593,89 @@ def codegen_collective(self, wrapper, output_name, input_names):
             f"group={output_name}_pg, "
             "async_op=True)"
         )
+
+
+class AllToAllSingle(CollectiveKernelAlloc):
+    """
+    NOTE: `all_to_all_single` is implemented differently from other collectives
+    because its output shape is data-dependent on input `output_split_sizes`.
+    Instead of letting Inductor manage allocation of output buffer that has
+    data-dependent dim size which is difficult to do, we resort to using
+    `ExternKernelAlloc` and letting the underlying `fun_col_impl._all_to_all_single`
+    kernel manage the output buffer allocation.
+    """
+
+    def __init__(
+        self,
+        layout,
+        inputs,
+        constant_args,
+        arg_index_to_input_index,
+    ):
+        super().__init__(layout, inputs, constant_args)
+        self.inputs = inputs
+        self.arg_index_to_input_index = arg_index_to_input_index
+
+    @classmethod
+    def create(
+        cls,
+        x: "TensorBox",
+        output_split_sizes: Optional["TensorBox"],
+        input_split_sizes: Optional["TensorBox"],
+        tag: str,
+        ranks: List[int],
+        group_size: int,
+    ):
+        x_realized = cls.realize_input(x)
+        inputs_allow_none = []
+        output_split_sizes_realized = None
+        if output_split_sizes is not None:
+            output_split_sizes_realized = cls.realize_input(output_split_sizes)
+        input_split_sizes_realized = None
+        if input_split_sizes is not None:
+            input_split_sizes_realized = cls.realize_input(input_split_sizes)
+        inputs_allow_none = [
+            x_realized,
+            output_split_sizes_realized,
+            input_split_sizes_realized,
+        ]
+        inputs = [inp for inp in inputs_allow_none if inp is not None]
+        arg_index_to_input_index = {}
+        for i in range(len(inputs_allow_none)):
+            if inputs_allow_none[i] is not None:
+                arg_index_to_input_index[i] = inputs.index(inputs_allow_none[i])
+            else:
+                arg_index_to_input_index[i] = None
+
+        return AllToAllSingle(
+            layout=inputs[0].get_layout(),
```

Comment:
Shouldn't this be the layout of the output?  I thought it needed data-dependent shapes?  This looks simpler than I expected.

Reply:
thanks - yes it's my oversight, I switched back to constructing output size with unbacked symint as first dim